crates/sui-types/src/base_types.rs

```diff
@@ -543,27 +543,25 @@ impl SuiAddress {
             .map(SuiAddress)
     }
 
-    /// A workaround to derive address with padded address_seed when converting it from BigInt to bytes.
-    pub fn legacy_try_from(authenticator: &ZkLoginAuthenticator) -> SuiResult<Self> {
+    /// This derives a zkLogin address by parising the iss and address_seed from [struct ZkLoginAuthenticator].
+    /// Define as iss_bytes_len || iss_bytes || padded_32_byte_address_seed. This is to be differentiated with
+    /// try_from_unpadded defined below.
+    pub fn try_from_padded(inputs: &ZkLoginInputs) -> SuiResult<Self> {
+        Ok((&PublicKey::from_zklogin_inputs(inputs)?).into())
+    }
+
+    /// Define as iss_bytes_len || iss_bytes || unpadded_32_byte_address_seed.
+    pub fn try_from_unpadded(inputs: &ZkLoginInputs) -> SuiResult<Self> {
```

Comment:
consider calling this `try_from_legacy_unpadded` to stress that is related to the other functions/variables with legacy in their name.

Reply:
so legacy == padded. ie what wallet did "incorrectly". 

i removed the term legacy from code since its rather confusing. its either padded or unpadded. wallet happened to use padded version in the past so it is "legacy"